src/main/java/io/xpring/xrpl/model/XRPTransaction.java

```diff
@@ -201,7 +235,34 @@ static XRPTransaction from(Transaction transaction) {
       }
     }
 
+    // Transactions report their timestamps since the Ripple Epoch, which is 946,684,800 seconds after
+    // the unix epoch. Convert transaction's timestamp to a unix timestamp.
+    // See: https://xrpl.org/basic-data-types.html#specifying-time
+    Optional<Integer> timestamp = Optional.empty();
```

Comment:
I think you want a `long` here since adding 946,684,800 + <arbitrary int value> could overflow. 

The (documentation also notes this](https://xrpl.org/basic-data-types.html#specifying-time):
> Don't convert Ripple Epoch times to UNIX Epoch times in 32-bit variables: this could lead to integer overflows.

Reply:
Note that JS has 2^52 values in a `number` so we're not too concerned there: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number

Reply:
wow, great catch.  I updated this to be a `Long`, but just to clarify, should it be a `Number`? or does `Long` work? (I know they both work, but stylistically...)

Reply:
I would use a long since most people will know the precision off the top of their heads. 

The number type I linked above is a JS class. I actually didn't know there was a [Number](https://docs.oracle.com/javase/8/docs/api/java/lang/Number.html) class in java and I certainly have never seen it used.